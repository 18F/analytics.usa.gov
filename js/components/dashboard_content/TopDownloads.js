import React, { useRef, useEffect, useState } from "react";
import PropTypes from "prop-types";
import d3 from "d3";

import ChartBuilder from "../../lib/chart_helpers/chart_builder";
import DataLoader from "../../lib/data_loader";
import barChart from "../../lib/chart_helpers/barchart";
import formatters from "../../lib/chart_helpers/formatters";
import FilterSelect from "../select/FilterSelect";

/**
 * Retrieves the top downloads report from the passed data URL and creates a
 * visualization for the count of users downloading files for the current
 * agency.
 *
 * @param {object} props the properties for the component
 * @param {string} props.agency the agency display name to be used in the chart
 * description.
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @param {number} props.numberOfListingsToDisplay the count of downloads
 * listings to display in the bar chart.
 * @returns {import('react').ReactElement} The rendered element
 */
function TopDownloads({ agency, dataHrefBase, numberOfListingsToDisplay }) {
  const reportFilters = [
    ["Yesterday", "top-downloads-yesterday"],
    ["7 Days", "top-downloads-7-days"],
    ["30 Days", "top-downloads-30-days"],
    ["90 Days", "top-downloads-90-days"],
  ];
  const [currentFilter, setCurrentFilter] = useState(reportFilters[0]);
  const [shouldDisplayDownloads, setShouldDisplayDownloads] = useState(true);
  const ref = useRef(null);

  useEffect(() => {
    const initChart = async () => {
      if (currentFilter) {
        await loadDataAndBuildChart();
      }
    };
    initChart().catch(console.error);
  }, [currentFilter]);

  async function loadDataAndBuildChart() {
    let data;

    try {
      data = await DataLoader.loadDailyReportJSON(
        `${dataHrefBase}/${currentFilter[1]}.json`,
      );
    } catch (e) {
      data = { totals: {} };
    }
    await buildChartForData(data);
  }

  async function buildChartForData(data) {
    let downloadData;
    if (data && data.data && Array.isArray(data.data) && data.data.length > 2) {
      downloadData = data;
      setShouldDisplayDownloads(true);
    } else {
      downloadData = { data: [] };
      setShouldDisplayDownloads(false);
    }

    const chartBuilder = new ChartBuilder();
    await chartBuilder
      .setElement(ref.current)
      .setData(downloadData)
      .setTransformer((d) => d.data.slice(0, numberOfListingsToDisplay))
      .setRenderer(
        barChart()
          .value((d) => +d.total_events)
          .label(
            (d) =>
              `<div class="grid-row">
                <div class="grid-col-12 text--overflow-ellipsis">
                  <a target="_blank" rel="noopener" href="http://${d.page}">
                    ${d.page_title}
                  </a>
                </div>
                <div class="grid-col-12">
                  <div class="grid-row text--overflow-wrap">
                    <div class="grid-col-auto text--lowercase dark-gray">
                      <span>${formatters.formatURL(d.page).trim()}</span><span class="padding-x-05">/</span>
                    </div>
                    <div class="grid-col-auto text--lowercase">
                      <a target="_blank" rel="noopener" href="${d.linkUrl}">
                        download file
                      </a>
                    </div>
                  </div>
                </div>
              </div>`,
          )
          .scale((values) =>
            d3.scale
              .linear()
              .domain([0, 1, d3.max(values)])
              .rangeRound([0, 1, 100]),
          )
          .format(formatters.addCommas),
      )
      .build();
  }

  async function filterChangeHandler(fileName) {
    if (!fileName) return;

    const selectedFilter = reportFilters.find((reportFilter) => {
      return reportFilter[1] == fileName;
    });
    await setCurrentFilter(selectedFilter);
  }

  function timeIntervalDescription() {
    if (currentFilter[0] == "Yesterday") {
      return currentFilter[0].toLowerCase();
    } else {
      return `over the last ${currentFilter[0].toLowerCase()}`;
    }
  }

  return (
    <>
      <p className="margin-top-0 margin-bottom-1">
        <em>
          {shouldDisplayDownloads
            ? `Top files downloaded ${timeIntervalDescription()} on ${agency} hostnames.`
            : `Top downloads data ${timeIntervalDescription()} is unavailable for ${agency} hostnames.`}
        </em>
      </p>
      <div className="grid-row">
        <div className="display-flex card:grid-col-12 mobile-lg:grid-col-fill card:flex-justify-center mobile-lg:flex-justify-start card:padding-bottom-105 mobile-lg:padding-bottom-0">
          {shouldDisplayDownloads && (
            <p className="margin-top-05 margin-bottom-05">
              <a
                href={`${dataHrefBase}/${currentFilter[1]}.csv`}
                aria-label={`${currentFilter[1]}.csv`}
              >
                Download the data
                <svg
                  className="usa-icon margin-bottom-neg-05 margin-left-05"
                  aria-hidden="true"
                  focusable="false"
                  role="img"
                >
                  <use xlinkHref="/assets/uswds/img/sprite.svg#file_present"></use>
                </svg>
              </a>
            </p>
          )}
        </div>
        <div className="card:grid-col-12 mobile-lg:grid-col-auto">
          <div className="display-flex card:flex-justify-center mobile-lg:flex-justify-end">
            <FilterSelect
              filters={reportFilters}
              defaultFilterValue={reportFilters[0][1] || ""}
              onChange={filterChangeHandler}
              name={"top downloads chart time filter"}
            />
          </div>
        </div>
      </div>
      {shouldDisplayDownloads && (
        <figure className="top-downloads__bar-chart" ref={ref}>
          <div className="data chart__bar-chart text--capitalize margin-top-2"></div>
        </figure>
      )}
    </>
  );
}

TopDownloads.propTypes = {
  agency: PropTypes.string.isRequired,
  dataHrefBase: PropTypes.string.isRequired,
  numberOfListingsToDisplay: PropTypes.number.isRequired,
};

export default TopDownloads;
