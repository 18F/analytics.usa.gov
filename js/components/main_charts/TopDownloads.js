import React, { useRef, useEffect } from "react";
import PropTypes from "prop-types";
import d3 from "d3";

import barChart from "../../lib/chart_helpers/barchart";
import formatters from "../../lib/chart_helpers/formatters";
import renderBlock from "../../lib/chart_helpers/renderblock";

/**
 * Retrieves the top downloads report from the passed data URL and creates a
 * visualization for the count of users downloading files for the current
 * agency.
 *
 * @param {String} dataHrefBase the URL of the base location of the data to be
 * downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 */
function TopDownloads({ dataHrefBase }) {
  const reportURL = `${dataHrefBase}/top-downloads-yesterday.json`;
  const ref = useRef(null);

  useEffect(() => {
    const initTopDownloads = async () => {
      const result = await d3
        .select(ref.current)
        .datum({
          source: reportURL,
          block: ref.current,
        })
        .call(
          renderBlock
            .loadAndRender()
            .transform((d) => d.data.slice(0, 10))
            .render(
              barChart()
                .value((d) => +d.total_events)
                .label((d) =>
                  [
                    '<span class="name"><a class="top-download-page" target="_blank" rel="noopener" href=http://',
                    d.page,
                    ">",
                    d.page_title,
                    "</a></span> ",
                    '<span class="domain" >',
                    formatters.formatURL(d.page),
                    "</span> ",
                    '<span class="divider">/</span> ',
                    '<span class="filename"><a class="top-download-file" target="_blank" aria-label="',
                    formatters.formatFile(d.file_name),
                    '" rel="noopener" href=',
                    formatters.formatProtocol(d.page),
                    formatters.formatFile(d.file_name),
                    ">",
                    "download file",
                    "</a></span>",
                  ].join(""),
                )
                .scale((values) =>
                  d3.scale
                    .linear()
                    .domain([0, 1, d3.max(values)])
                    .rangeRound([0, 1, 100]),
                )
                .format(formatters.addCommas),
            ),
        );
      return result;
    };
    initTopDownloads().catch(console.error);
  }, []);

  return (
    <figure id="top-downloads" ref={ref}>
      <div className="data bar-chart"></div>
    </figure>
  );
}

TopDownloads.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default TopDownloads;
