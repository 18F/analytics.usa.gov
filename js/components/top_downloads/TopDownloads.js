import React, { useRef, useEffect } from "react";
import PropTypes from "prop-types";
import d3 from "d3";

import barChart from "../../lib/chart_helpers/barchart";
import formatters from "../../lib/chart_helpers/formatters";
import renderBlock from "../../lib/chart_helpers/renderblock";

function TopDownloads({ dataURL, dataPrefix }) {
  const reportURL = `${dataURL}/${dataPrefix}/top-downloads-yesterday.json`;
  const ref = useRef(null);

  useEffect(() => {
    const initTopDownloads = async () => {
      const result = await d3
        .select(ref.current)
        .datum({
          source: reportURL,
          block: ref.current,
        })
        .call(
          renderBlock
            .loadAndRender()
            .transform((d) => d.data.slice(0, 10))
            .render(
              barChart()
                .value((d) => +d.total_events)
                .label((d) =>
                  [
                    '<span class="name"><a class="top-download-page" target="_blank" rel="noopener" href=http://',
                    d.page,
                    ">",
                    d.page_title,
                    "</a></span> ",
                    '<span class="domain" >',
                    formatters.formatURL(d.page),
                    "</span> ",
                    '<span class="divider">/</span> ',
                    '<span class="filename"><a class="top-download-file" target="_blank" aria-label="',
                    formatters.formatFile(d.file_name),
                    '" rel="noopener" href=',
                    formatters.formatProtocol(d.page),
                    formatters.formatFile(d.file_name),
                    ">",
                    "download file",
                    "</a></span>",
                  ].join(""),
                )
                .scale((values) =>
                  d3.scale
                    .linear()
                    .domain([0, 1, d3.max(values)])
                    .rangeRound([0, 1, 100]),
                )
                .format(formatters.addCommas),
            ),
        );
      return result;
    };
    initTopDownloads().catch(console.error);
  }, []);

  return (
    <figure id="top-downloads" ref={ref}>
      <div className="data bar-chart"></div>
    </figure>
  );
}

TopDownloads.propTypes = {
  dataURL: PropTypes.string.isRequired,
  dataPrefix: PropTypes.string.isRequired,
};

export default TopDownloads;
