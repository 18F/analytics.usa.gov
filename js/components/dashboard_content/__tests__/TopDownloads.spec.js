import React from "react";
import { render, screen, waitFor } from "@testing-library/react";

import DownloadReportFactory from "../../../../spec/factories/reports/download";
import { delay } from "../../../../spec/support/test_utilities";
import DataLoader from "../../../lib/data_loader";
import TopDownloads from "../TopDownloads";

jest.mock("../../../lib/data_loader", () => ({
  ...jest.requireActual("../../../lib/data_loader"),
  loadDailyReportJSON: jest.fn(),
}));

describe("TopDownloads", () => {
  let component;
  let data;

  describe("when data is not loaded", () => {
    beforeEach(async () => {
      DataLoader.loadDailyReportJSON.mockImplementation(() => {
        return Promise.resolve(null);
      });
      component = render(
        <TopDownloads
          dataHrefBase="http://www.example.com/data/"
          agency="Interior"
          numberOfListingsToDisplay={5}
        />,
      );
      await waitFor(() =>
        screen.getByText(
          "Top downloads data yesterday is unavailable for Interior hostnames.",
        ),
      );
    });

    it("renders a component in loading state", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe("when data is loaded", () => {
    beforeEach(async () => {
      data = DownloadReportFactory.build(
        {},
        { transient: { dataItemCount: 20 } },
      );

      data.data[0].file_name =
        "https://travel.state.gov/dv 2026 plain language instructions and faqs.pdf";

      DataLoader.loadDailyReportJSON.mockImplementation(() => {
        return Promise.resolve(data);
      });
      component = render(
        <TopDownloads
          dataHrefBase="http://www.example.com/data/"
          agency="Interior"
          numberOfListingsToDisplay={5}
        />,
      );
      await waitFor(() => screen.getByText(data.data[0].page));
      // Wait for barchart transition animation to complete (200 ms, set in
      // js/lib/chart_helpers/barchart.js)
      await delay(300);
    });

    it("renders a component with data loaded", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe("when data loading has an error", () => {
    const error = "you broke it";

    beforeEach(async () => {
      console.error = jest.fn();
      DataLoader.loadDailyReportJSON.mockImplementation(() => {
        return Promise.reject(error);
      });
      component = render(
        <TopDownloads
          dataHrefBase="http://www.example.com/data/"
          agency="Interior"
          numberOfListingsToDisplay={5}
        />,
      );
      await waitFor(() =>
        screen.getByText(
          "Top downloads data yesterday is unavailable for Interior hostnames.",
        ),
      );
    });

    it("renders a component in error state", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
