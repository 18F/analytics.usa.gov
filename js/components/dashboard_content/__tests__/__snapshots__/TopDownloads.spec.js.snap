// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopDownloads when data is loaded renders a component with data loaded 1`] = `
<DocumentFragment>
  <p
    class="margin-top-0 margin-bottom-1"
  >
    <em>
      Top files downloaded yesterday on Interior hostnames.
    </em>
  </p>
  <div
    class="grid-row"
  >
    <div
      class="display-flex card:grid-col-12 mobile-lg:grid-col-fill card:flex-justify-center mobile-lg:flex-justify-start card:padding-bottom-105 mobile-lg:padding-bottom-0"
    >
      <p
        class="margin-top-05 margin-bottom-05"
      >
        <a
          aria-label="top-downloads-yesterday.csv"
          href="http://www.example.com/data//top-downloads-yesterday.csv"
        >
          Download the data
          <svg
            aria-hidden="true"
            class="usa-icon margin-bottom-neg-05 margin-left-05"
            focusable="false"
            role="img"
          >
            <use
              xlink:href="/assets/uswds/img/sprite.svg#file_present"
            />
          </svg>
        </a>
      </p>
    </div>
    <div
      class="card:grid-col-12 mobile-lg:grid-col-auto"
    >
      <div
        class="display-flex card:flex-justify-center mobile-lg:flex-justify-end"
      >
        <select
          aria-label="top downloads chart time filter"
          class="usa-select chart__filter margin-top-0"
        >
          <option
            value="top-downloads-yesterday"
          >
            Yesterday
          </option>
          <option
            value="top-downloads-7-days"
          >
            7 Days
          </option>
          <option
            value="top-downloads-30-days"
          >
            30 Days
          </option>
          <option
            value="top-downloads-90-days"
          >
            90 Days
          </option>
          <option
            value="top-downloads-current-year"
          >
            Current Year
          </option>
          <option
            value="top-downloads-current-fiscal-year"
          >
            Current Fiscal Year
          </option>
          <option
            value="top-downloads-previous-year"
          >
            Previous Year
          </option>
          <option
            value="top-downloads-previous-fiscal-year"
          >
            Previous Fiscal Year
          </option>
        </select>
      </div>
    </div>
  </div>
  <figure
    class="top-downloads__bar-chart"
  >
    <div
      class="data chart__bar-chart text--capitalize margin-top-2"
    >
      <div
        class="chart__bar-chart__item margin-bottom-2"
      >
        <div
          class="chart__bar-chart__item__label-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__label dark-gray text--overflow-ellipsis grid-col-fill padding-right-1"
          >
            <div
              class="grid-row"
            >
              
                
              <div
                class="grid-col-12 text--overflow-ellipsis"
              >
                
                  
                <a
                  href="http://robust-childhood.com"
                  rel="noopener"
                  target="_blank"
                >
                  
                    deploy virtual synergies
                  
                </a>
                
                
              </div>
              
                
              <div
                class="grid-col-12"
              >
                
                  
                <div
                  class="grid-row text--overflow-wrap"
                >
                  
                    
                  <div
                    class="grid-col-auto text--lowercase dark-gray"
                  >
                    
                      
                    <span>
                      robust-childhood.com
                    </span>
                    <span
                      class="padding-x-05"
                    >
                      /
                    </span>
                    
                    
                  </div>
                  
                    
                  <div
                    class="grid-col-auto text--lowercase"
                  >
                    
                      
                    <a
                      href="https://dizzy-ladybug.net"
                      rel="noopener"
                      target="_blank"
                    >
                      
                        download file
                      
                    </a>
                    
                    
                  </div>
                  
                  
                </div>
                
                
              </div>
              
              
            </div>
          </div>
          <div
            class="chart__bar-chart__item__value grid-col-auto text--align-right"
          >
            3,048
          </div>
        </div>
        <div
          class="chart__bar-chart__item__bar-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__bar bg-palette-color-5 margin-top-0 grid-col-12"
            style="width: 46%;"
          />
        </div>
      </div>
      <div
        class="chart__bar-chart__item margin-bottom-2"
      >
        <div
          class="chart__bar-chart__item__label-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__label dark-gray text--overflow-ellipsis grid-col-fill padding-right-1"
          >
            <div
              class="grid-row"
            >
              
                
              <div
                class="grid-col-12 text--overflow-ellipsis"
              >
                
                  
                <a
                  href="http://early-locality.org"
                  rel="noopener"
                  target="_blank"
                >
                  
                    leverage back-end paradigms
                  
                </a>
                
                
              </div>
              
                
              <div
                class="grid-col-12"
              >
                
                  
                <div
                  class="grid-row text--overflow-wrap"
                >
                  
                    
                  <div
                    class="grid-col-auto text--lowercase dark-gray"
                  >
                    
                      
                    <span>
                      early-locality.org
                    </span>
                    <span
                      class="padding-x-05"
                    >
                      /
                    </span>
                    
                    
                  </div>
                  
                    
                  <div
                    class="grid-col-auto text--lowercase"
                  >
                    
                      
                    <a
                      href="https://questionable-lake.net"
                      rel="noopener"
                      target="_blank"
                    >
                      
                        download file
                      
                    </a>
                    
                    
                  </div>
                  
                  
                </div>
                
                
              </div>
              
              
            </div>
          </div>
          <div
            class="chart__bar-chart__item__value grid-col-auto text--align-right"
          >
            6,120
          </div>
        </div>
        <div
          class="chart__bar-chart__item__bar-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__bar bg-palette-color-5 margin-top-0 grid-col-12"
            style="width: 91%;"
          />
        </div>
      </div>
      <div
        class="chart__bar-chart__item margin-bottom-2"
      >
        <div
          class="chart__bar-chart__item__label-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__label dark-gray text--overflow-ellipsis grid-col-fill padding-right-1"
          >
            <div
              class="grid-row"
            >
              
                
              <div
                class="grid-col-12 text--overflow-ellipsis"
              >
                
                  
                <a
                  href="http://wide-intervention.name"
                  rel="noopener"
                  target="_blank"
                >
                  
                    orchestrate efficient applications
                  
                </a>
                
                
              </div>
              
                
              <div
                class="grid-col-12"
              >
                
                  
                <div
                  class="grid-row text--overflow-wrap"
                >
                  
                    
                  <div
                    class="grid-col-auto text--lowercase dark-gray"
                  >
                    
                      
                    <span>
                      wide-intervention.name
                    </span>
                    <span
                      class="padding-x-05"
                    >
                      /
                    </span>
                    
                    
                  </div>
                  
                    
                  <div
                    class="grid-col-auto text--lowercase"
                  >
                    
                      
                    <a
                      href="https://kindhearted-opening.biz"
                      rel="noopener"
                      target="_blank"
                    >
                      
                        download file
                      
                    </a>
                    
                    
                  </div>
                  
                  
                </div>
                
                
              </div>
              
              
            </div>
          </div>
          <div
            class="chart__bar-chart__item__value grid-col-auto text--align-right"
          >
            4,440
          </div>
        </div>
        <div
          class="chart__bar-chart__item__bar-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__bar bg-palette-color-5 margin-top-0 grid-col-12"
            style="width: 66%;"
          />
        </div>
      </div>
      <div
        class="chart__bar-chart__item margin-bottom-2"
      >
        <div
          class="chart__bar-chart__item__label-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__label dark-gray text--overflow-ellipsis grid-col-fill padding-right-1"
          >
            <div
              class="grid-row"
            >
              
                
              <div
                class="grid-col-12 text--overflow-ellipsis"
              >
                
                  
                <a
                  href="http://vain-coil.name"
                  rel="noopener"
                  target="_blank"
                >
                  
                    mesh frictionless networks
                  
                </a>
                
                
              </div>
              
                
              <div
                class="grid-col-12"
              >
                
                  
                <div
                  class="grid-row text--overflow-wrap"
                >
                  
                    
                  <div
                    class="grid-col-auto text--lowercase dark-gray"
                  >
                    
                      
                    <span>
                      vain-coil.name
                    </span>
                    <span
                      class="padding-x-05"
                    >
                      /
                    </span>
                    
                    
                  </div>
                  
                    
                  <div
                    class="grid-col-auto text--lowercase"
                  >
                    
                      
                    <a
                      href="https://crisp-barber.com"
                      rel="noopener"
                      target="_blank"
                    >
                      
                        download file
                      
                    </a>
                    
                    
                  </div>
                  
                  
                </div>
                
                
              </div>
              
              
            </div>
          </div>
          <div
            class="chart__bar-chart__item__value grid-col-auto text--align-right"
          >
            494
          </div>
        </div>
        <div
          class="chart__bar-chart__item__bar-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__bar bg-palette-color-5 margin-top-0 grid-col-12"
            style="width: 8%;"
          />
        </div>
      </div>
      <div
        class="chart__bar-chart__item margin-bottom-2"
      >
        <div
          class="chart__bar-chart__item__label-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__label dark-gray text--overflow-ellipsis grid-col-fill padding-right-1"
          >
            <div
              class="grid-row"
            >
              
                
              <div
                class="grid-col-12 text--overflow-ellipsis"
              >
                
                  
                <a
                  href="http://shameful-contest.org"
                  rel="noopener"
                  target="_blank"
                >
                  
                    engineer world-class networks
                  
                </a>
                
                
              </div>
              
                
              <div
                class="grid-col-12"
              >
                
                  
                <div
                  class="grid-row text--overflow-wrap"
                >
                  
                    
                  <div
                    class="grid-col-auto text--lowercase dark-gray"
                  >
                    
                      
                    <span>
                      shameful-contest.org
                    </span>
                    <span
                      class="padding-x-05"
                    >
                      /
                    </span>
                    
                    
                  </div>
                  
                    
                  <div
                    class="grid-col-auto text--lowercase"
                  >
                    
                      
                    <a
                      href="https://open-territory.name"
                      rel="noopener"
                      target="_blank"
                    >
                      
                        download file
                      
                    </a>
                    
                    
                  </div>
                  
                  
                </div>
                
                
              </div>
              
              
            </div>
          </div>
          <div
            class="chart__bar-chart__item__value grid-col-auto text--align-right"
          >
            6,711
          </div>
        </div>
        <div
          class="chart__bar-chart__item__bar-wrapper grid-row"
        >
          <div
            class="chart__bar-chart__item__bar bg-palette-color-5 margin-top-0 grid-col-12"
            style="width: 100%;"
          />
        </div>
      </div>
    </div>
  </figure>
</DocumentFragment>
`;

exports[`TopDownloads when data is not loaded renders a component in loading state 1`] = `
<DocumentFragment>
  <p
    class="margin-top-0 margin-bottom-1"
  >
    <em>
      Top downloads data yesterday is unavailable for Interior hostnames.
    </em>
  </p>
  <div
    class="grid-row"
  >
    <div
      class="display-flex card:grid-col-12 mobile-lg:grid-col-fill card:flex-justify-center mobile-lg:flex-justify-start card:padding-bottom-105 mobile-lg:padding-bottom-0"
    />
    <div
      class="card:grid-col-12 mobile-lg:grid-col-auto"
    >
      <div
        class="display-flex card:flex-justify-center mobile-lg:flex-justify-end"
      >
        <select
          aria-label="top downloads chart time filter"
          class="usa-select chart__filter margin-top-0"
        >
          <option
            value="top-downloads-yesterday"
          >
            Yesterday
          </option>
          <option
            value="top-downloads-7-days"
          >
            7 Days
          </option>
          <option
            value="top-downloads-30-days"
          >
            30 Days
          </option>
          <option
            value="top-downloads-90-days"
          >
            90 Days
          </option>
          <option
            value="top-downloads-current-year"
          >
            Current Year
          </option>
          <option
            value="top-downloads-current-fiscal-year"
          >
            Current Fiscal Year
          </option>
          <option
            value="top-downloads-previous-year"
          >
            Previous Year
          </option>
          <option
            value="top-downloads-previous-fiscal-year"
          >
            Previous Fiscal Year
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TopDownloads when data loading has an error renders a component in error state 1`] = `
<DocumentFragment>
  <p
    class="margin-top-0 margin-bottom-1"
  >
    <em>
      Top downloads data yesterday is unavailable for Interior hostnames.
    </em>
  </p>
  <div
    class="grid-row"
  >
    <div
      class="display-flex card:grid-col-12 mobile-lg:grid-col-fill card:flex-justify-center mobile-lg:flex-justify-start card:padding-bottom-105 mobile-lg:padding-bottom-0"
    />
    <div
      class="card:grid-col-12 mobile-lg:grid-col-auto"
    >
      <div
        class="display-flex card:flex-justify-center mobile-lg:flex-justify-end"
      >
        <select
          aria-label="top downloads chart time filter"
          class="usa-select chart__filter margin-top-0"
        >
          <option
            value="top-downloads-yesterday"
          >
            Yesterday
          </option>
          <option
            value="top-downloads-7-days"
          >
            7 Days
          </option>
          <option
            value="top-downloads-30-days"
          >
            30 Days
          </option>
          <option
            value="top-downloads-90-days"
          >
            90 Days
          </option>
          <option
            value="top-downloads-current-year"
          >
            Current Year
          </option>
          <option
            value="top-downloads-current-fiscal-year"
          >
            Current Fiscal Year
          </option>
          <option
            value="top-downloads-previous-year"
          >
            Previous Year
          </option>
          <option
            value="top-downloads-previous-fiscal-year"
          >
            Previous Fiscal Year
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
